electrum/gui/qml/components/InvoiceDialog.qml

```diff
@@ -216,10 +217,23 @@ ElDialog {
 
                             BtcField {
                                 id: amountBtc
+                                Layout.preferredWidth: amountFontMetrics.advanceWidth('0') * 14 + leftPadding + rightPadding
                                 fiatfield: amountFiat
-                                enabled: !amountMax.checked
+                                readOnly: amountMax.checked
+                                color: readOnly
+                                    ? Material.accentColor
+                                    : Material.foreground
                                 onTextAsSatsChanged: {
-                                    invoice.amountOverride = textAsSats
+                                    if (!amountMax.checked)
+                                        invoice.amountOverride.satsInt = textAsSats.satsInt
+                                }
+                                Connections {
+                                    target: invoice.amountOverride
+                                    function onSatsIntChanged() {
+                                        console.log('amuontOverride satsIntChanged, sats=' + invoice.amountOverride.satsInt)
+                                        if (amountMax.checked)  // amountOverride updated by max amount estimate
+                                            amountBtc.text = Config.formatSats(invoice.amountOverride.satsInt)
```

Comment:
Buggy if you have `config.BTC_AMOUNTS_ADD_THOUSANDS_SEP` enabled. Toggle max on then off, and try to edit the amount but with some spaces still left in the field. It does not work properly when you try to create the tx (incorrect amount).

Needs to use Config.formatSatsForEditing?